### ISSUE TITLE
TypeError: float argument required, not NoneType

### ISSUE THREAD

**Issue by shibumi** [whipper-team/whipper#402]
Whipper version: 0.7.3
Host: Arch Linux

Log:
```
❯ WHIPPER_DEBUG=DEBUG WHIPPER_LOGFILE=whipper.log whipper cd rip
CDDB disc id: 52050806
MusicBrainz disc id gYSLVmN5HKcAUJu00T1de5fw5Ds-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+6+96800+150+16188+33255+51557+67395+82170&tracks=6&id=gYSLVmN5HKcAUJu00T1de5fw5Ds-
Disc duration: 00:21:28.666, 6 audio tracks

Matching releases:

Artist  : FLiP
Title   : DEAR GIRLS
Duration: 00:21:26.000
URL     : https://musicbrainz.org/release/2acfe703-c1d5-4aa2-a80f-7ca8fb3a7191
Release : 2acfe703-c1d5-4aa2-a80f-7ca8fb3a7191
Type    : EP
Barcode : 4562104046316
Cat no  : DFCL-1599

Peak level: 0.982849                     
Rip quality: 100.00%
Peak level: 0.988525                     
Rip quality: 100.00%
Peak level: 0.982849                     
Rip quality: 100.00%
Peak level: 0.982849                     
Rip quality: 100.00%
Traceback (most recent call last):       
  File "/usr/lib/python2.7/site-packages/whipper/extern/task/task.py", line 518, in c
    callable(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/whipper/program/cdparanoia.py", line 316, in _read
    self._done()
  File "/usr/lib/python2.7/site-packages/whipper/program/cdparanoia.py", line 392, in _done
    self.quality = self._parser.getTrackQuality()
  File "/usr/lib/python2.7/site-packages/whipper/program/cdparanoia.py", line 200, in getTrackQuality
    raise RuntimeError("cdparanoia couldn't read any frames "
RuntimeError: cdparanoia couldn't read any frames for the current track
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/__init__.py", line 868, in emit
    msg = self.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 741, in format
    return fmt.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 465, in format
    record.message = record.getMessage()
  File "/usr/lib/python2.7/logging/__init__.py", line 329, in getMessage
    msg = msg % self.args
TypeError: float argument required, not NoneType
Logged from file program.py, line 512
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/__init__.py", line 868, in emit
    msg = self.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 741, in format
    return fmt.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 465, in format
    record.message = record.getMessage()
  File "/usr/lib/python2.7/logging/__init__.py", line 329, in getMessage
    msg = msg % self.args
TypeError: float argument required, not NoneType
Logged from file program.py, line 514
Peak level: 0.988525                             
Rip quality: 100.00%
Peak level: 0.982849                             
Rip quality: 100.00%
track  1: rip accurate     (max confidence      1) v1 [cb8d9eee], v2 [8b098dd0], DB [cb8d9eee, 8b098dd0]
track  2: rip accurate     (max confidence      1) v1 [d91a6231], v2 [b9e6b321], DB [d91a6231, b9e6b321]
track  3: rip accurate     (max confidence      1) v1 [2687411b], v2 [63951bb8], DB [2687411b, 63951bb8]
track  4: rip accurate     (max confidence      1) v1 [0700013b], v2 [0bdebe32], DB [0700013b, 0bdebe32]
track  5: rip accurate     (max confidence      1) v1 [80a394f8], v2 [8dd31dc2], DB [80a394f8, 8dd31dc2]
track  6: rip accurate     (max confidence      1) v1 [516a67ad], v2 [ffd0426e], DB [516a67ad, ffd0426e]
```

Whipper debug log: https://paste.xinu.at/mFZ0k/

**Comment by JoeLametta** [comment]
Unfortunately I was away and couldn't manage to download the linked debug log. Could you please post an updated link?

Cheers,
Joe


P.S.: How come whipper seems to fail on track 5 but at the end of the shortlog I it querying the AccurateRip DB against all tracks (track 5 and 6 too)?

**Comment by JoeLametta** [comment]
Any reply? (in case of no reply this issue is going to be closed in a week).

**Comment by shibumi** [comment]
Sorry for the late reply, I have no idea. If you wish I could test it again with whipper version 8.0

**Comment by JoeLametta** [comment]
Thanks for the reply!

> If you wish I could test it again with whipper version 8.0

As I'm going to tag whipper v0.9.0 very soon (surely this week), I'll suggest testing with that one once it's available. :wink:

**Comment by shibumi** [comment]
@JoeLametta thanks, i will do.

**Comment by shibumi** [comment]
Hi,
I can confirm this bugs are fixed in version 0.9.0.

```
❯ WHIPPER_DEBUG=DEBUG WHIPPER_LOGFILE=whipper.log whipper cd rip
CDDB disc id: 52050806
MusicBrainz disc id gYSLVmN5HKcAUJu00T1de5fw5Ds-
MusicBrainz lookup URL https://musicbrainz.org/cdtoc/attach?toc=1+6+96800+150+16188+33255+51557+67395+82170&tracks=6&id=gYSLVmN5HKcAUJu00T1de5fw5Ds-
Disc duration: 00:21:28.666, 6 audio tracks

Matching releases:

Artist  : FLiP
Title   : DEAR GIRLS
Duration: 00:21:26.000
URL     : https://musicbrainz.org/release/2acfe703-c1d5-4aa2-a80f-7ca8fb3a7191
Release : 2acfe703-c1d5-4aa2-a80f-7ca8fb3a7191
Type    : EP
Barcode : 4562104046316
Cat no  : DFCL-1599

Peak level: 0.982849                     
Rip quality: 100.00%
Peak level: 0.988525                     
Rip quality: 100.00%
Peak level: 0.982849                     
Rip quality: 100.00%
Peak level: 0.982849                     
Rip quality: 100.00%
Peak level: 0.988525                     
Rip quality: 100.00%
Peak level: 0.982849                     
Rip quality: 100.00%
track  1: rip accurate     (max confidence      1) v1 [cb8d9eee], v2 [8b098dd0], DB [cb8d9eee, 8b098dd0]
track  2: rip accurate     (max confidence      1) v1 [d91a6231], v2 [b9e6b321], DB [d91a6231, b9e6b321]
track  3: rip accurate     (max confidence      1) v1 [2687411b], v2 [63951bb8], DB [2687411b, 63951bb8]
track  4: rip accurate     (max confidence      1) v1 [0700013b], v2 [0bdebe32], DB [0700013b, 0bdebe32]
track  5: rip accurate     (max confidence      1) v1 [80a394f8], v2 [8dd31dc2], DB [80a394f8, 8dd31dc2]
track  6: rip accurate     (max confidence      1) v1 [516a67ad], v2 [ffd0426e], DB [516a67ad, ffd0426e]
'WHIPPER_DEBUG=DEBUG WHIPPER_LOGFILE=whipper.log whipper cd rip' time: 11:22.94s, cpu: 7%
```

Cheers,
Chris

**Comment by JoeLametta** [comment]
All right, thanks!
